### PR TITLE
Biddle Verbs + Spatial Gridmapping + Init optimizations [PORT]

### PR DESCRIPTION
**Pull request by Bibby0110**
# Document the changes in your pull request

Ports the following for performance. Chances of working: 0.2%
- https://github.com/tgstation/tgstation/pull/69376
- https://github.com/tgstation/tgstation/pull/69703
- https://github.com/tgstation/tgstation/pull/69079
- https://github.com/tgstation/tgstation/pull/69638
- https://github.com/tgstation/tgstation/pull/69742
- https://github.com/tgstation/tgstation/pull/65589
- https://github.com/tgstation/tgstation/pull/63118

- https://github.com/tgstation/tgstation/pull/60219
- https://github.com/tgstation/tgstation/pull/61422
- https://github.com/tgstation/tgstation/pull/63765

# Wiki Documentation

# Changelog

:cl:  Multitude of brave TG coders, ported by TheGamerdk
experimental: Many performance tweaks
/:cl:

### CODE REVIEW

File: code/game/objects/items/devices/radio/radio.dm [hunk 1]
```diff
@@ -481,6 +481,8 @@
 
 	var/mob/living/silicon/robot/R = loc
 	if(istype(R))
+		if(!R.module)
+			return
 		for(var/ch_name in R.module.radio_channels)
```
Comment (redmoogle) [inline code/game/objects/items/devices/radio/radio.dm:622]: why not a `?.`

Comment (Bibby0110) [reply code/game/objects/items/devices/radio/radio.dm:622]: lazy

File: code/game/objects/items/devices/radio/radio.dm [hunk 2]
```diff
@@ -6,46 +8,98 @@
...
+	var/translate_binary = FALSE
+	/// If true, can say/hear on the special CentCom channel.
+	var/independent = FALSE
+	/// If true, hears all well-known channels automatically, and can say/hear on the Syndicate channel.
+	var/syndie = FALSE
+	/// associative list of the encrypted radio channels this radio is currently set to listen/broadcast to, of the form: list(channel name = TRUE or FALSE)
+	var/list/channels
+	/// associative list of the encrypted radio channels this radio can listen/broadcast to, of the form: list(channel name = channel frequency)
 	var/list/secure_radio_connections
-	var/list/radio_sounds = list('yogstation/sound/effects/radio1.ogg','yogstation/sound/effects/radio2.ogg','yogstation/sound/effects/radio3.ogg')
```
Comment (redmoogle) [inline code/game/objects/items/devices/radio/radio.dm:57]: Accidental removal

**Comment by redmoogle** [comment]
Another thing I noticed is that assemblies with the voice analyzer dont seem to work (unless I dont know how to turn them on or they were just broken)

**Comment by ToasterBiome** [comment]
will SSspeech_controller play nice with our snowflake say handling?

**Comment by redmoogle** [comment]
> will SSspeech_controller play nice with our snowflake say handling?

works just fine

**Comment by Bibby0110** [comment]
> will SSspeech_controller play nice with our snowflake say handling?

We have snowflake say handling?

**Comment by Bibby0110** [comment]
stale

**Pull request closed without merging**